gym_ai2thor/tasks.py

```diff
@@ -93,10 +112,226 @@ def transition_reward(self, state):
+
+        # always last word of the sentence. Has to be spelled exactly
+        self.curr_object_type = self.curr_instruction.split(' ')[-1]
+        print('Current instruction: {}. object type (last word in sentence): {} '.format(
+            self.curr_instruction, self.curr_object_type))
+
+        return self.curr_instruction
+
+
+class NaturalLanguageLookAtObjectTask(NaturalLanguageBaseTask):
+    """
+    This task consists of requiring the agent to get close to the object type and look at it
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # how far the target object is from the agent to receive reward
+        self.distance_threshold_3d = kwargs['task'].get('distance_threshold_3d', 1.0)
+        # self.terminal_if_lookat_wrong_object # todo hard to refactor and not worth it? but cozmo had it this way. how to define what the wrong object is? Impossible? So rare too.
+
+    def transition_reward(self, event):
+        reward, done = self.movement_reward, False
+        # check if current target object is in middle of screen and close
+        target_objs = check_if_focus_and_close_enough_to_object_type(event, self.curr_object_type,
+                                                 distance_threshold_3d=self.distance_threshold_3d)
+        if target_objs > 0:
+            print('Stared at {} and is close enough. Num objects in view and '
+                  'close: {}'.format(self.curr_object_type, target_objs))
+            reward += self.default_reward
+            done = True
+
+        return reward, done
+
+    def reset(self):
+        return super(NaturalLanguageLookAtObjectTask, self).reset()
+
+
+class NaturalLanguageNavigateToObjectTask(NaturalLanguageBaseTask):
+    """
+    This task consists of requiring the agent to get close to the object type and look at it
+    The closeness is set by distance_threshold=0.84
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def transition_reward(self, event):
+        reward, done = self.movement_reward, False
+        # check if current target object is in middle of screen and close.
+        # Closer than NaturalLanguageLookAtObjectTask
+        target_objs = check_if_focus_and_close_enough_to_object_type(event,
+                                                                event.metadata['curr_object_type'],
+                                                                distance_threshold_3d=0.84)
+        if target_objs > 0:
+            print('Stared at {} and is close enough. Num objects in view and '
+                  'close: {}'.format(self.curr_object_type, target_objs))
+            reward += self.default_reward
+            done = True
+
+        return reward, done
+
+    def reset(self):
+        return super(NaturalLanguageNavigateToObjectTask, self).reset()
+
+
+class NaturalLanguagePickUpObjectTask(NaturalLanguageBaseTask):
```

Comment:
I would restructure this to have this class inherit from both NaturalLanguageBaseTask and PickupTask. We can discuss this in slack